### PR TITLE
Replace boost::this_thread::sleep by std::this_thread::sleep_for

### PR DESCRIPTION
**Pull request by SunBlack**

### CODE REVIEW

**Comment by SergioRAgostinho** [comment]
This is concurrent work with I was doing in #2700. But let's go with this, because it is done.

File: test/io/test_grabbers.cpp [hunk 1]
```diff
@@ -126,7 +128,7 @@ TEST (PCL, ImageGrabberTIFF)
     size_t niter = 0;
     while (!signal_received)
     {
-      boost::this_thread::sleep (boost::posix_time::microseconds (10000));
+      std::this_thread::sleep_for(100ms);
```
Comment (taketwo) [inline test/io/test_grabbers.cpp]:
```suggestion
      std::this_thread::sleep_for(10ms);
```

Comment (taketwo) [reply test/io/test_grabbers.cpp]: And in other occurrences below.

**Comment by SunBlack** [comment]
> This is concurrent work with I was doing in #2700. But let's go with this, because it is done.

Sorry, I just remembered that you are working one replaying `mutex` from `boost` to `std`. but forgot that you wanted to change `this_thread::sleep` too. After my last both Boost PR I just looked for remaining `posix_time` occurrences (there are still some left).

File: visualization/tools/pcd_viewer.cpp [hunk 2]
```diff
@@ -731,7 +735,7 @@ main (int argc, char** argv)
         }
         p->spinOnce ();
       }
-      boost::this_thread::sleep (boost::posix_time::microseconds (100));
+	  std::this_thread::sleep_for(100us);
```
Comment (SergioRAgostinho) [inline visualization/tools/pcd_viewer.cpp]: Indentation got messed up here.

**Comment by SergioRAgostinho** [comment]
Everyone agrees on a big squash?

**Pull request merged**